push-sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/apns/ApnsUtil.java

```diff
@@ -66,4 +68,29 @@ public static String readDefaultTopic(final byte[] keystore, final char[] passwo
 
         return null; // if no topic was found we try with null
     }
+
+    public static boolean checkValidity(final byte[] keystore, final char[] password) {
+        try {
+            final KeyStore keyStore = KeyStore.getInstance(KEYSTORE_TYPE);
+            keyStore.load(new ByteArrayInputStream(keystore), password);
+
+            final Enumeration<String> aliases = keyStore.aliases();
+
+            while (aliases.hasMoreElements()) {
+                final String alias = aliases.nextElement();
+                final X509Certificate certificate = (X509Certificate) keyStore.getCertificate(alias);
+
+                try{
+                    certificate.checkValidity();
+                } catch (CertificateExpiredException | CertificateNotYetValidException e) {
+                    LOGGER.error("Provided APNs .p12 file is expired or not yet valid");
```

Comment:
So, this can - depending on the actual amount of push apptempts cause some mega logs, in case the customer/user has invalid push - and does not bother to update

Reply:
But somehow, I feel this should be logged ... since it is an issue that requires action from the owner of the certificate

Reply:
By default the log looks like:

```

14:52:21,237 DEBUG [org.jboss.aerogear.unifiedpush.message.cache.SimpleApnsClientCache] (Thread-1 (ActiveMQ-client-global-threads)) no cached connection for e0351314-c5a4-4b52-beef-e8e8626edb5c-dev, establishing it
14:52:21,242 ERROR [org.jboss.aerogear.unifiedpush.message.sender.apns.ApnsUtil] (Thread-1 (ActiveMQ-client-global-threads)) Provided APNs .p12 file is expired
14:52:21,242 ERROR [org.jboss.aerogear.unifiedpush.message.sender.apns.PushyApnsSender] (Thread-1 (ActiveMQ-client-global-threads)) Unable to send notifications, client is not connected. Removing from cache pool
14:52:21,242 WARN  [org.jboss.aerogear.unifiedpush.message.NotificationDispatcher] (Thread-1 (ActiveMQ-client-global-threads)) Error on 'ios' delivery: Unable to send notifications, client is not connected
14:52:21,251 DEBUG [org.jboss.aerogear.unifiedpush.message.cache.SimpleApnsClientCache] (Thread-1 (ActiveMQ-client-global-threads)) Removed client from cache for e0351314-c5a4-4b52-beef-e8e8626edb5c
```

for each invocation, w/ an invalid certificate

Reply:
Of course `DEBUG` is only in our own Docker image enabled.

so, each would do 2 ERROR and one WARN

Reply:
Would it be better to check the certificate in the NotificationDispatcher: https://github.com/aerogear/aerogear-unifiedpush-server/blob/master/push-sender/src/main/java/org/jboss/aerogear/unifiedpush/message/NotificationDispatcher.java#L66 ?

But at this point we would have to check if we are dealing with APNS, then cast the sender and the variant and then do the check.